[CI] RankDocsSortBuilderTests testEqualsAndHashcode failing

**Build Scans:**
- [elasticsearch-intake #9211 / part1](https://gradle-enterprise.elastic.co/s/fbpk3yiercawo)
- [elasticsearch-pull-request #29202 / part-1](https://gradle-enterprise.elastic.co/s/mt4zds5tfw2zy)

**Reproduction Line:**
```
./gradlew ":server:test" --tests "org.elasticsearch.search.retriever.rankdoc.RankDocsSortBuilderTests.testEqualsAndHashcode" -Dtests.seed=A6C11B49B417857 -Dtests.locale=vi -Dtests.timezone=SystemV/EST5EDT -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testEqualsAndHashcode'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.search.retriever.rankdoc.RankDocsSortBuilderTests'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: RankDocsSortBuilder mutation should not be equal to original
Expected: not <org.elasticsearch.search.retriever.rankdoc.RankDocsSortBuilder@56f5b212>
     but: was <org.elasticsearch.search.retriever.rankdoc.RankDocsSortBuilder@56f5b212>
```

**Issue Reasons:**
- [main] 2 failures in test testEqualsAndHashcode (0.5% fail rate in 405 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -121,9 +121,6 @@ tests:
 - class: org.elasticsearch.xpack.ml.integration.MlJobIT
   method: testDeleteJobAsync
   issue: https://github.com/elastic/elasticsearch/issues/112212
-- class: org.elasticsearch.search.retriever.rankdoc.RankDocsSortBuilderTests
-  method: testEqualsAndHashcode
-  issue: https://github.com/elastic/elasticsearch/issues/112312
 - class: org.elasticsearch.search.retriever.RankDocRetrieverBuilderIT
   method: testRankDocsRetrieverWithCollapse
   issue: https://github.com/elastic/elasticsearch/issues/112254

--- a/server/src/test/java/org/elasticsearch/search/retriever/rankdoc/RankDocsSortBuilderTests.java
+++ b/server/src/test/java/org/elasticsearch/search/retriever/rankdoc/RankDocsSortBuilderTests.java
@@ -44,7 +44,7 @@ public class RankDocsSortBuilderTests extends AbstractSortTestCase<RankDocsSortB
     @Override
     protected RankDocsSortBuilder mutate(RankDocsSortBuilder original) throws IOException {
         RankDocsSortBuilder mutated = new RankDocsSortBuilder(original);
-        mutated.rankDocs(randomRankDocs(original.rankDocs().length + randomInt(100)));
+        mutated.rankDocs(randomRankDocs(original.rankDocs().length + randomIntBetween(10, 100)));
         return mutated;
     }
 

